商品詳細表示の実装

miracletea opened the pull request:
# What
商品詳細表示機能の作成

# Why
商品詳細表示を実装のため

【Gyazo】
・ ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画
https://gyazo.com/1a8102d111181782b1f7a01f20fbd914
・グイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画
https://gyazo.com/e0ca20172e3b501a25b6be9aa4b6d668
・ログアウト状態で、商品詳細ページへ遷移した動画
https://gyazo.com/c3e86e13072cb67686f46663a6b06c58

miracletea committed:
商品詳細画面の実装

yuri240 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,107 @@
+  <% end %>
+
+    <div class="item-explain-box">
+      <span><%= "商品説明" %></span>
```
37行目の商品説明も修正しましょう！

yuri240 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,107 @@
+  </div>
+  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
+  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
+  <%# //詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
詳細ページ最下部の以下記述においても保存した商品のカテゴリー名が表示されるようにしましょう。
また、実装後は上下のコメントアウトの削除もお忘れなくお願いします。

`<%='商品のカテゴリー名'%>をもっと見る`

yuri240 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,107 @@
+
+  <% if user_signed_in? && current_user.id != @item.user_id %>
+    <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
+  <% end %>
```
```ruby
<% if user_signed_in? && current_user.id != @item.user_id %>
```
上記32行目の記述は、
26行目と反対の意味になりますので、`else`に代替できるかと存じます。

`else`を使用することで、記述量が減り、リファクタリングに繋がりますので、ご参考にしていただけると幸いです。

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。

miracletea committed:
商品説明の修正

app/views/items/show.html.erb
  <% end %>

    <div class="item-explain-box">
      <span><%= "商品説明" %></span>
@yuri240 yuri240 6 minutes ago
37行目の商品説明も修正しましょう！

miracletea committed:
ページ最下部のカテゴリー表示を修正

app/views/items/show.html.erb
Comment on lines +102 to +104
  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
  <%# //詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
@yuri240 yuri240 9 minutes ago
詳細ページ最下部の以下記述においても保存した商品のカテゴリー名が表示されるようにしましょう。
また、実装後は上下のコメントアウトの削除もお忘れなくお願いします

miracletea committed:
if文法の修正

app/views/items/show.html.erb
Comment on lines +26 to +34
  <% if user_signed_in? && current_user.id == @item.user_id %>
      <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
      <p class="or-text">or</p>
      <%= link_to "削除", "#", data: {turbo_method: :delete}, class:"item-destroy" %>
  <% end %>
  <% if user_signed_in? && current_user.id != @item.user_id %>
    <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
  <% end %>
@yuri240 yuri240 10 minutes ago
<% if user_signed_in? && current_user.id != @item.user_id %>
上記32行目の記述は、
26行目と反対の意味になりますので、elseに代替できるかと存じます。

elseを使用することで、記述量が減り、リファクタリングに繋がりますので、ご参考にしていただけると幸いです。

miracletea committed:
if文のend忘れを修正

miracletea commented:
ご確認ありがとうございます。
コメント内容を反映しました、宜しくお願い致します。
【Gyazo】
・ ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画
https://gyazo.com/7cb6bb2143de2ec311202e9c0de4dad0
・ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画
https://gyazo.com/b0a21e89a29eac7dbb94edcb96ce3544
・ログアウト状態で、商品詳細ページへ遷移した動画
https://gyazo.com/cdd3b756a11d4f2b5307c260b92cfbc9

maimai0401 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,107 @@
+      後ろの商品 ＞
+    </a>
+  </div>
+  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
「〜しましょう」で終わる実装案内コメントは実装完了後に削除しましょう。
企業提出時に実装案内コメントが残っていないようにするためです。

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください。
どうぞよろしくおねがいします。

miracletea committed:
不要コメントの削除

miracletea commented:
ご確認ありがとうございます。
コメント内容を反映しました、宜しくお願い致します。

maimai0401 commented:
レビュー依頼ありがとうございます。

こちら`LGTM`とさせていただきます！実装お疲れさまでした。

【注意】
```ruby
①（次のブランチを切る前に）プルリクエストのMergeとmainブランチへのPullを忘れないようにしましょう。
②次のフェーズに進む際は必ず新しいブランチを作ってから作業しましょう。
```

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
引き続き頑張りましょう！応援しております！
